bindinfo/cache.go

```diff
@@ -83,8 +91,13 @@ func NewBindCacheUpdater(ctx sessionctx.Context, handle *Handle, parser *parser.
 }
 
 // NewHandle creates a Handle with a cache.
-func NewHandle() *Handle {
-	handle := &Handle{}
+func NewHandle(ctx sessionctx.Context) *Handle {
+	exec, _ := ctx.(sqlexec.SQLExecutor)
```

Comment:
`exec` and `ctx` is same thing. You cannot avoid data race of them by a different name.

Reply:
Handle has a sync.Mutex

Reply:
It just protects the `exec`. But it is the same as `ctx`, which is also used in 
cache updater.

Reply:
Cache updater only read, on write operator.

Reply:
It does not matter. It has internal stats, which will brings data race.

Reply:
ok